### ISSUE TITLE
Reproducing the experiment in Figure 2

### ISSUE THREAD

**Issue by jyhong0304** [IBM/concept_transformer#5]
First of all, your work is very interesting, and thank you very much for posting your repo!

I want to ask how to reproduce the experiment in Figure 2 in your paper.
I tried to reproduce the experiments of your model with 200 training examples using the following command:
``` python ctc_mnist.py --n_train_samples 200 --expl_lambda 2.0```

But I got only random guessing test accuracy, 0.507.
I missed something, so please let me know how to do it.
I would appreciate any help you can provide.

Regards,
Jinung Hong

**Comment by matrig** [comment]
Thank you very much for your interest in our work, Jinung!
You are right, the default `learning_rate` and `batch_size` parameters of the script are not suited for the MNIST experiment. I now include a snippet in the README indicating better hyperparamenter which should actually work as advertised:
`python ctc_mnist.py --learning_rate 0.0004 --max_epochs 150 --warmup 20 --batch_size 32 --n_train_samples 200 --expl_lambda 2.0`.
Let me know if these don't work better.
I also pushed the file [run_mnist_scaling_experiments.py](https://github.com/IBM/concept_transformer/blob/main/run_mnist_scaling_experiments.py) which will run `ctc_mnist.py` for varying number of samples to reproduce the whole sweep of Figure 2 (to use it you will have to add an additional dependency for logging with `pip install epyc`).
I hope this helps!

**Comment by jyhong0304** [comment]
Thank you so much for your comment and your help.
It worked perfectly!

Regards,
Jinyung Hong
